app/views/goals/index.html.erb

```diff
@@ -3,20 +3,24 @@
   <%= link_to '<<Back', user_goals_path(current_user) %>
 </div>
 
-This Week's Turing Community Totals:
+<h3>This Week's Turing Community Totals</h3>
 <hr>
-<div class="total-progress-bar">
-  <img class='icons' src= "<%= @sweat.image_path%>"> XXXXXX <%= @sweat_progress %> / <%= @sweat_total %>
-</div>
-
-<div class="total-progress-bar">
-  <img class='icons' src= "<%= @nutrition.image_path%>"> XXXXXX <%= @nutrition_progress %> / <%= @nutrition_total%>
-</div>
-
-<div class="total-progress-bar">
-  <img class='icons' src= "<%= @sleep.image_path%>"> XXXXXX <%= @sleep_progress %> /  <%= @sleep_total %> <br>
-</div>
 
-<div class="total-progress-bar">
-  <img class='icons' src= "<%= @personal.image_path%>"> XXXXXX <%= @personal_progress %> / <%= @personal_total %>
+<%@categories.each do |category| %>
+  <div class="goal_<%= category.name %>">
+    <div class='goal-title-icon'>
+      <img class='icons' src= "<%= category.image_path%>"> &nbsp;
+      <h3><%= category.name %></h3>
+    </div>
+    <div class="progress">
+      <% if category.get_community_total_count(current_week.id) %>
+        <div id="cat-<%=category.id%>", class="progress-bar" role="progressbar" style="width: <%=((category.get_community_progress(current_week.id) / category.get_community_total_count(current_week.id).to_f) * 100).round%>%;"
```

Comment:
this is also a great deal of calculation for the view to be responsible for. create a helper method to extract this logic or put it in your "Category" model 👍